### ISSUE TITLE
SSH master connection

### ISSUE THREAD

**Issue by nicolamos** [RainMark/oxfs#23]
I tried to connect to a server for which I have a master session running. It seems that offs does not use the SSH configuration that is available. Would it be possible to direct offs to use that connection?

**Comment by RainMark** [comment]
> I tried to connect to a server for which I have a master session running. It seems that offs does not use the SSH configuration that is available. Would it be possible to direct offs to use that connection?

You mean use ssh ControlMaster？

**Comment by nicolamos** [comment]
Yes exactly, or in any case use the SSH configuration a user might have (eg. ~/.ssh/config) for a given host. I have a server that I can access through SSH using a ProxyJump and all that is specified in my configuration file.

**Comment by RainMark** [comment]
> Yes exactly, or in any case use the SSH configuration a user might have (eg. ~/.ssh/config) for a given host. I have a server that I can access through SSH using a ProxyJump and all that is specified in my configuration file.

oxfs use paramiko to create ssh connection, but paramiko not support fully ssh configurations. https://github.com/paramiko/paramiko/issues/852
Do you have some suggestions?

**Comment by nicolamos** [comment]
I see. It seems that on this issue the development was stuck. 
Would it be possible to bypass the use of paramiko? Or directing it to use an already opened connection?

**Comment by RainMark** [comment]
> I see. It seems that on this issue the development was stuck. Would it be possible to bypass the use of paramiko? Or directing it to use an already opened connection?

maybe need some extra works, but i have not time to do this currently
